include/nigiri/routing/search.h

```diff
@@ -136,13 +138,27 @@ struct search {
     state_.starts_.clear();
     add_start_labels(q_.start_time_, true);
 
+    auto processing_start_time = std::chrono::system_clock::now();
+    auto timeout_reached = [=, this]() -> bool {
+      if (timeout_) {
+        return (std::chrono::system_clock::now() - processing_start_time) >= *timeout_;
+      }
+
+      return false;
+    };
+
     while (true) {
       trace("start_time={}\n", search_interval_);
 
       search_interval();
 
       if (is_ontrip() || max_interval_reached() ||
-          n_results_in_interval() >= q_.min_connection_count_) {
+          n_results_in_interval() >= q_.min_connection_count_ ||
+          timeout_reached()) {
+        if (timeout_reached()) {
+          trace("  stopping search because timeout was reached\n");
```

Comment:
I think it's easier to put this into the trace blow. Then it's sufficient to grep for `finished` to find the reason it's finished (besides being more compact code - less tracing code makes the actual logic more readable).